lib-multisrc/mangabox/src/eu/kanade/tachiyomi/multisrc/mangabox/MangaBox.kt

```diff
@@ -338,53 +405,77 @@ abstract class MangaBox(
-        Pair("29", "Sci fi"),
-        Pair("30", "Seinen"),
-        Pair("31", "Shoujo"),
-        Pair("32", "Shoujo ai"),
-        Pair("33", "Shounen"),
-        Pair("34", "Shounen ai"),
-        Pair("35", "Slice of life"),
-        Pair("36", "Smut"),
-        Pair("37", "Sports"),
-        Pair("38", "Supernatural"),
-        Pair("39", "Tragedy"),
-        Pair("40", "Webtoons"),
-        Pair("41", "Yaoi"),
-        Pair("42", "Yuri"),
+        Pair("action", "Action"),
+        Pair("adult", "Adult"),
+        Pair("adventure", "Adventure"),
+        Pair("comedy", "Comedy"),
+        Pair("cooking", "Cooking"),
+        Pair("doujinshi", "Doujinshi"),
+        Pair("drama", "Drama"),
+        Pair("ecchi", "Ecchi"),
+        Pair("fantasy", "Fantasy"),
+        Pair("gender-bender", "Gender bender"),
+        Pair("harem", "Harem"),
+        Pair("historical", "Historical"),
+        Pair("horror", "Horror"),
+        Pair("isekai", "Isekai"),
+        Pair("josei", "Josei"),
+        Pair("manhua", "Manhua"),
+        Pair("manhwa", "Manhwa"),
+        Pair("martial-arts", "Martial arts"),
+        Pair("mature", "Mature"),
+        Pair("mecha", "Mecha"),
+        Pair("medical", "Medical"),
+        Pair("mystery", "Mystery"),
+        Pair("one-shot", "One shot"),
+        Pair("psychological", "Psychological"),
+        Pair("romance", "Romance"),
+        Pair("school-life", "School life"),
+        Pair("sci-fi", "Sci fi"),
+        Pair("seinen", "Seinen"),
+        Pair("shoujo", "Shoujo"),
+        Pair("shoujo-ai", "Shoujo ai"),
+        Pair("shounen", "Shounen"),
+        Pair("shounen-ai", "Shounen ai"),
+        Pair("slice-of-life", "Slice of life"),
+        Pair("smut", "Smut"),
+        Pair("sports", "Sports"),
+        Pair("supernatural", "Supernatural"),
+        Pair("tragedy", "Tragedy"),
+        Pair("webtoons", "Webtoons"),
+        Pair("yaoi", "Yaoi"),
+        Pair("yuri", "Yuri"),
     )
 
-    // To be overridden if using tri-state genres
-    protected open fun getAdvancedGenreFilters(): List<AdvGenre> = emptyList()
-
     open class UriPartFilter(displayName: String, private val vals: Array<Pair<String?, String>>) :
         Filter.Select<String>(displayName, vals.map { it.second }.toTypedArray()) {
         fun toUriPart() = vals[state].first
     }
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        ListPreference(screen.context).apply {
+            key = PREF_USE_MIRROR
+            title = "Mirror"
+            entries = mirrorEntries
+            entryValues = mirrorEntries.map { "${URL_PREFIX}$it" }.toTypedArray()
+            setDefaultValue(entryValues[0]) // Set default value
+
+            // Set initial value from SharedPreferences or default value
+            val initialValue = preferences.getString(PREF_USE_MIRROR, mirrorEntries[0])
+            value = initialValue
+            summary = initialValue!!.removePrefix(URL_PREFIX)
```

Comment:
not needed? "%s" should work for summary

Reply:
You can set url on init instead, (where you check if current url isn't in the mirror list)